src/index.js

```diff
@@ -1,225 +1,64 @@
-const telemetry = require('./telemetry')
-
-function decodeBlock(block, car) {
-  const codec = codecs[block.cid.code]
-
-  if (!codec) {
-    logger.error({ car }, `Unsupported codec ${block.cid.code} in the block at offset ${block.blockOffset}`)
-    throw new Error(`Unsupported codec ${block.cid.code} in the block at offset ${block.blockOffset}`)
-  }
-
-  // Decoded the block data
-  const data = codec.decode(block.data)
-
-  if (codec.label.startsWith('dag')) {
-    const { type, blocks } = decodeUnixFs(data.Data)
-
-    data.Data = { type, blocks }
+const { logger, serializeError } = require('./lib/logging')
+const telemetry = require('./lib/telemetry')
+const { storeCar } = require('./lib/car')
+
+/**
+ * Event to invoke the lambda
+ * @typedef {Object} Event
+ * @property {Array<Record>} Records
+ */
+
+/**
+ * @typedef {Object} Record
+ * @property {String} body - because of SQS message format, body can be the car id or the JSON stringified record with following properties:
+ * - skipExists - default `false`, skip to process the CAR if already exists in db
+ * - decodeBlocks - default `false`, decode CAR blocks
+ * @example { body: 'car-id' }
+ * @example { body: '{"body":"car-id",skipExists:true,"decodeBlocks":true}' }
+ */
+
+function parseEvent(event) {
+  if (event.Records.length !== 1) {
+    logger.error(`Indexer Lambda invoked with ${event.Records.length} CARs while should be 1`)
+    throw new Error(`Indexer Lambda invoked with ${event.Records.length} CARs while should be 1`)
   }
 
-  return data
-}
-
-async function updateCarStatus(carId, block, car) {
-  const currentPosition = block.offset + block.length
-
-  /*
-    The condition is needed if another block, next to this one,
-    has been evaluated first. This can happen due to hwp induced parallelism.
-  */
-  return writeDynamoItem(
-    false,
-    carsTable,
-    'path',
-    carId,
-    {
-      currentPosition
-    },
-    car,
-    {
-      currentPosition: ['<', currentPosition]
+  const body = event.Records[0].body
+  if (body[0] === '{') {
+    try {
+      const { body: carId, skipExists, decodeBlocks } = JSON.parse(body)
+      return { carId, skipExists, decodeBlocks }
+    } catch {
+      throw new Error('Invalid JSON in event body: ' + body)
     }
-  )
-}
-
-async function storeNewBlock(carId, type, block, car, data = {}) {
-  for (const link of data?.Links ?? []) {
-    link.Hash = link.Hash.toString()
   }
-
-  const cid = cidToKey(block.cid)
-
-  await writeDynamoItem(true, blocksTable, primaryKeys.blocks, cid, {
-    type,
-    /* c8 ignore next */
-    createdAt: now || new Date().toISOString(),
-    cars: [{ car: carId, offset: block.blockOffset, length: block.blockLength }],
-    data
-  }, car)
-
-  if (skipPublishing) {
-    return
-  }
-
-  return publishToSQS(publishingQueue, cid, carId)
+  return { carId: event.Records[0].body }
 }
 
+/**
+ * Returns an empty object to signal we have consumed all the messages
```

Comment:
Is it possible for it to return and _not_ have consumed all the messages?

Reply:
the comment is outdated, the lambda consumes one car at time, throwing an error or an empty object

Reply:
Hmm, yes I guess my question is why do we need to return _anything_ if in the failure case we throw?